Adds Local Ledger Readme (#1035)
algorand/indexer: docs/LocalLedger.md

# Local Ledger

> In order to ensure that various operations are performed in a timely manner, the Indexer stores a "local ledger" in the supplied data directory.  This ledger allows for fast lookup of data and significantly reduces the amount of time to perform initial loading of the state data.

Reviewer: ```suggestion
In order to ensure that various operations are performed in a timely manner, the Indexer stores a "local ledger" in the supplied data directory.  This ledger allows for fast lookup of data and significantly reduces the amount of time to perform initial loading of the state data. In previous versions of indexer the database was queried to fetch this data.
```

> To maintain backwards compatability with the current Indexer API, the Indexer also maintains a database backend that is used to service the REST API.  Because of this, the local ledger must have data "migrated" to it whenever the Indexer is started.  A simplified flow is shown below:

Reviewer: ```suggestion
In case of upgrading from an earlier version of Indexer, or setting up a new Indexer server with an existing database, the local ledger must sometimes be initialized when the Indexer is started.  A simplified flow is shown below:
```

```mermaid
graph TD;
    A[Start Indexer]-->B{Do files exist in the data directory?};
    B -- No --> C([See Troubleshooting Below]);
    B -- Yes --> D{Is it caught up to db?};
    D -- Yes --> E([Indexer Starts]);
    D -- No --> F[Indexer performs migration];
    F --> E;
```

> Migration can be performed in a couple of ways:

Reviewer: ```suggestion
Initialization can be performed in a couple of ways:
```

1) The Indexer can be told to use a catchpoint to "fast-forward" through the process.
2) The Indexer can use a "sequential scan" of the database and insert data into the ledger one by one.

By default, the Indexer will choose option 2 unless otherwise specified.

> ## Troubleshooting

Reviewer: @shiqizng does the user need to manually create the ledger file?

> ## Troubleshooting

Reviewer 2: No, user doesn't need to manually create the files. I removed that check from the PR. @AlgoStephenAkiki
